### PR TITLE
Text nodes and mixed content formatting overhaul

### PR DESCRIPTION
**Pull request by datho7561**
Try to improve formatting handling of text content and mixed content.

Clsoes redhat-developer/vscode-xml#377

### CODE REVIEW

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLFormatter.java [hunk 1]
```diff
@@ -41,6 +41,7 @@
 import org.eclipse.lsp4j.Position;
 import org.eclipse.lsp4j.Range;
 import org.eclipse.lsp4j.TextEdit;
+import org.jsoup.helper.StringUtil;
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLFormatter.java:44]: Please use StringUtils from lemminx.

**Comment by datho7561** [comment]
It looks like I broke all the new formatting features when I was trying to fix regressions. This will need a lot more work.

**Comment by datho7561** [comment]
FormatOverhaul

This is the case that is a regression: comments after text content are put on a new line instead of staying at the end of the line:

**Comment by datho7561** [comment]
Closing this PR, since it doesn't work and doesn't clean up the code that much. We likely need to do a larger refactoring in order to improve the formatter.

**Pull request closed without merging**